nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/pcap/SplitPCAP.java

```diff
@@ -0,0 +1,244 @@
+                                    out.write(templatePcap.readBytesFull());
+                                    splitFilesList.add(newFlowFile);
+
+                                } catch (IOException e) {
+                                    throw new IOException(e.getMessage());
+                                }
+
+                                loadedPackets.clear();
+                                loadedPackets.add(currentPacket);
+                                currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;
+                                templatePcap.packets().clear();
+                            }
+
+                            loadedPackets.add(currentPacket);
+                            currentPcapTotalLength += currentPacket.totalLength();
+                        }
+
+                        // If there are any packets left over, create a new flowfile.
+                        if (!loadedPackets.isEmpty()) {
+                            templatePcap.packets().addAll(loadedPackets);
+                            FlowFile newFlowFile = session.create(flowFile);
+                            try (final OutputStream out = session.write(newFlowFile)) {
+                                out.write(templatePcap.readBytesFull());
+                                splitFilesList.add(newFlowFile);
+
+                            } catch (IOException e) {
+                                throw new IOException(e.getMessage());
+                            }
+                        }
+                    } catch (IOException e) {
+                        throw new ProcessException(e.getMessage());
+                    }
+                }
+            });
+        } catch (ProcessException e) {
+            session.remove(splitFilesList);
+            splitFilesList.clear();
+            session.putAttribute(flowFile, ERROR_REASON_LABEL, e.getMessage());
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final String fragmentId = UUID.randomUUID().toString();
+        final String originalFileName = flowFile.getAttribute(CoreAttributes.FILENAME.key());
+        final String originalFileNameWithoutExtension = originalFileName.substring(0, originalFileName.lastIndexOf("."));
+
+        IntStream.range(0, splitFilesList.size()).forEach(index -> {
+            FlowFile split = splitFilesList.get(index);
+            Map<String, String> attributes = new HashMap<>();
+            attributes.put(CoreAttributes.FILENAME.key(), originalFileNameWithoutExtension + "-" + index + ".pcap");
+            attributes.put(FRAGMENT_COUNT, String.valueOf(splitFilesList.size()));
+            attributes.put(FRAGMENT_ID, fragmentId);
+            attributes.put(FRAGMENT_INDEX, Integer.toString(index));
+            attributes.put(SEGMENT_ORIGINAL_FILENAME, originalFileName);
+            session.putAllAttributes(split, attributes);
+        });
```

Comment:
I learnt from SplitJson that it not necessary to have the attributes Map instantiated anew for each iteration. Its enough to have one instance and override those entries which are unique to the split file.

```suggestion
        final String fragmentId = UUID.randomUUID().toString();
        final String originalFileName = flowFile.getAttribute(CoreAttributes.FILENAME.key());
        final String originalFileNameWithoutExtension = originalFileName.substring(0, originalFileName.lastIndexOf("."));
        final Map<String, String> attributes = new HashMap<>();
        attributes.put(FRAGMENT_COUNT, String.valueOf(splitFilesList.size()));
        attributes.put(FRAGMENT_ID, fragmentId);
         attributes.put(SEGMENT_ORIGINAL_FILENAME, originalFileName);
         
        IntStream.range(0, splitFilesList.size()).forEach(index -> {
            FlowFile split = splitFilesList.get(index);
            attributes.put(CoreAttributes.FILENAME.key(), originalFileNameWithoutExtension + "-" + index + ".pcap");
            attributes.put(FRAGMENT_INDEX, Integer.toString(index));  
            session.putAllAttributes(split, attributes);
        });
```